doc/configuration.md

```diff
@@ -0,0 +1,247 @@
+Configuration
+=============
+The Datadog tracer's configuration can be specified in multiple ways:
+
+- programmatically in C++ code via the `TracerOptions` object defined in
+  [datadog/opentracing.h][1],
+- process-wide via setting values for certain [environment variables][4],
+- [dynamically][3] by reading JSON-formatted text, as in done in the [nginx
+  plugin][2].
+
+Most options support all three methods of configuration.
+
+Environment variables override any corresponding configuration in
+`TracerOptions` or loaded from JSON.
+
+Options
+-------
+### Agent Host
+The name of the host at which the Datadog Agent can be contacted.
```

Comment:
I think that it could be either.  Wasn't sure how to say that.  Perhaps I should be more specific: "Either the network name of the host at which the Datadog Agent can be contacted, or the host's IP address."